src/linter/ui5Types/SourceFileLinter.ts

```diff
@@ -325,6 +326,43 @@ export default class SourceFileLinter {
 				messageDetails: this.#messageDetails ? formatMessage(MESSAGES.DETAILS__LIB_INIT_2) : undefined,
 			});
 		}
+
+		if (initArg) {
+			this.#analyzeLibInitDeprecatedLibs(initArg);
+		}
+	}
+
+	#analyzeLibInitDeprecatedLibs(initArg: ts.ObjectLiteralExpression) {
+		const dependenciesNode = initArg.properties.find((prop) => {
+			return ts.isPropertyAssignment(prop) &&
+				ts.isIdentifier(prop.name) &&
+				prop.name.text === "dependencies";
+		});
+
+		if (!dependenciesNode ||
+			!ts.isPropertyAssignment(dependenciesNode) ||
+			!ts.isArrayLiteralExpression(dependenciesNode.initializer)) {
+			return;
+		}
+
+		dependenciesNode.initializer.elements.forEach((dependency) => {
+			if (!ts.isStringLiteral(dependency)) {
+				// We won't be interested if the elements of the Array are not of type
+				// StringLiteral, so we ignore such cases here (if such at all).
+				return;
+			}
+
+			const curLibName = dependency.text;
+
+			if (deprecatedLibraries.includes(curLibName)) {
+				this.#reporter.addMessage({
+					ruleId: RULES["ui5-linter-no-deprecated-api"],
```

Comment:
why `ui5-linter-no-deprecated-api`, when there is `ui5-linter-no-deprecated-library`? The same message is currently used for different ruleIds. Do we need to discuss the meaning of ruleIds? I would expect that the same ID is used for the same kind of check (e.g. for a deprecated library), regardless of where it is found (e.g. in an API call as in here, or in a ui5.yaml, or .library).

`ui5-linter-no-deprecated-api` I rather understand as a finding where the called API itself is deprecated, which is not the case in here.

Reply:
Yeah, that's correct! I've missed the deprecated-library rule.